kernel/core/utils/stringtools.hpp

```diff
@@ -343,34 +354,48 @@ struct NumberFormatsVoter
     /////////////////////////////////////////////////
     int getFormat()
     {
-        // As soon as there are more than 2 set, we will have at least one bad sample that would be parsed wrong
-        int inval_cnt = 0;
-        for(int i = 0; i < FMT_COUNT; i++)
-        {
-            if(m_num_format_votes[i] > 0)
-                inval_cnt++;
-        }
-        if(inval_cnt > 2 || m_num_format_votes[FMT_INVALID] > 0)
+        if(m_num_format_votes[FMT_INVALID] > 0)
             return NUM_INVALID;
 
+        int cnt_set = 0;
         int num_format = 0;
-        if((m_num_format_votes[FMT_K_EU] > m_num_format_votes[FMT_K_US]) && (m_num_format_votes[FMT_K_EU] > m_num_format_votes[FMT_K_SPACE]))
+        if(m_num_format_votes[FMT_K_EU] > 0)
+        {
             num_format |= NUM_K_EU;
-        else if((m_num_format_votes[FMT_K_US] > m_num_format_votes[FMT_K_EU]) && (m_num_format_votes[FMT_K_US] > m_num_format_votes[FMT_K_SPACE]))
+            cnt_set++;
+        }
+        if(m_num_format_votes[FMT_K_US] > 0)
+        {
             num_format |= NUM_K_US;
-        else if(m_num_format_votes[FMT_K_SPACE] > 0)
+            cnt_set++;
+        }
+        if(m_num_format_votes[FMT_K_SPACE] > 0)
+        {
             num_format |= NUM_K_SPACE;
+            cnt_set++;
+        }
+
+        if(cnt_set > 1)
+            return NUM_INVALID;
 
-        int add = 0;
-        if(num_format == 0)
-            add = 1;
+        // IF we have some ambigious votes that correspont to unique decimals use these to set thousands seperator.
+        if(m_num_format_votes[FMT_DEC_EU] > 0)
+        {
+            num_format |= NUM_DECIMAL_EU | (m_num_format_votes[FMT_AMBIGIOUS_DOT] > 0 ? NUM_K_EU : 0);

```

Comment:
vote for these ambiguous cases and if at least one unique DECIMAL separator was found and also ambiguous thousands separators where found, set also the thousand separator.

Reply:
That depends on how easy that would be implemented. Consider that a fancy solution might be hard to understand and therefore hard to maintain. If the ambiguous votes do not really impact performance or change correctness, I'd vote for keeping them.